src/js/environment/environment.ts

```diff
@@ -20,18 +20,19 @@ import {
 import { browserVersion } from './browser-version';
 import { osVersion } from './os-version';
 import { memoize } from 'utils/underscore';
+import { GenericObject } from 'types/generic.type';
 
-const userAgent = navigator.userAgent;
+const userAgent: string = navigator.userAgent;
 
-function supportsPassive() {
+function supportsPassive(): boolean {
     let passiveOptionRead = false;
 
     try {
-        const opts = Object.defineProperty({}, 'passive', {
+        const opts: GenericObject = Object.defineProperty({}, 'passive', {
             get: () => (passiveOptionRead = true)
         });
-        window.addEventListener('testPassive', null, opts);
-        window.removeEventListener('testPassive', null, opts);
+        window.addEventListener('testPassive', {} as EventListenerObject, opts);
```

Comment:
hmmm, does this not let `null` in based on the TS "standard types"?

Reply:
Correct!

Reply:
Looks like this should be a no-op function. i get the following when passing an `{}` as the func arg in the browser
```
document.getElementById('testtest').addEventListener('click', {}, {});
TypeError: Property 'handleEvent' is not callable.
TypeError: Property 'handleEvent' is not callable.
```

Reply:
https://github.com/Modernizr/Modernizr/issues/1894

Reply:
ts noop added.